### ISSUE TITLE
MongoDB errors return invalid HTTP status codes

### ISSUE THREAD

**Issue by dukeofsussex** [pencilblue/pencilblue#1189]
**Description:**
Example case:
Opening a page that uses the db and passes "bad" sorting options (old syntax `order: [{'publish_date': pb.DAO.DESC}, {'created': pb.DAO.DESC}]`) to the db query results in the error page being shown with the status code `2`.

Response:
```
code:2,
message:"bad sort specification",
stack:"MongoError: bad sort specification at ...
```
A similar issue has been fixed in #1160. 

For the 1.0 rework it might be good to not return mongodb error codes as HTTP status codes, as they can cause all sorts of problems (in my case it was a proxy not letting the response through).

PencilBlue Version: 0.7.1

Node.js Version: 4.4.5

Environment Description: Win Dev

**Comment by brianhyder** [comment]
Yeah, we started using a "code" property on the errors to set status code.  Then when we updated the MongoDB driver they had started doing the same thing.  We'll consider changing this for 1.0 to alleviate this issue.
